### DISCUSSION TITLE
PnP Search Result - Display address links from quick links web part (

### DISCUSSION THREAD

**Discussion by KaveSys** [microsoft-search/pnp-modern-search#4071] [Q&A]
Hello,
I don't see any other discussion related to this so I'm trying to achieve this thing.
I want to show also the links (in quick links web part or somewhere else) in search results when a word is search.

I'm using this query text on Search Results:

{searchTerms} Path:https://xxxxxxxxx.sharepoint.com/sites/URLSite AND (contentclass:STS_ListItem OR contentclass:STS_List OR contentclass:STS_List_DocumentLibrary -Path:https://xxxxxxxxx.sharepoint.com/sites/URLSite/Shared%20Documents* AND -Path:https://xxxxxxxxx.sharepoint.com/sites/URLSite/Lists* AND -Path:https://xxxxxxxxx.sharepoint.com/sites/URLSite/SitePages* AND -Path:https://xxxxxxxxx.sharepoint.com/sites/URLSite/SiteAssets* AND -Path:https://xxxxxxxxx.sharepoint.com/sites/URLSite/Style%20Library* AND -Path:https://xxxxxxxxx.sharepoint.com/sites/URLSite/FormServerTemplates AND -Title:"XXXX XXXX XXXX") -filetype:aspx

Maybe I'm doing something wrong? I try to search for a word in an address link but it doesn't show up on search results.

**Comment by kasperbolarsen** [comment]
Sorry, but I just don't understand what you are trying to do? 
It looks like you want results from a specific site collection, and it has to be 
a list item
OR
a List
OR 
a Document Library

and then you are excluding a number of libraries and stuff

There might be an issue with the syntax, perhaps this is what you are aiming for?
{searchTerms} (Path:https://xxxxxxxxx.sharepoint.com/sites/URLSite AND (contentclass:STS_ListItem OR contentclass:STS_List OR contentclass:STS_List_DocumentLibrary))
-Path:https://xxxxxxxxx.sharepoint.com/sites/URLSite/Shared%20Documents* 
-Path:https://xxxxxxxxx.sharepoint.com/sites/URLSite/Lists* 
-Path:https://xxxxxxxxx.sharepoint.com/sites/URLSite/SitePages* 
-Path:https://xxxxxxxxx.sharepoint.com/sites/URLSite/SiteAssets* 
-Path:https://xxxxxxxxx.sharepoint.com/sites/URLSite/Style%20Library* 
-Path:https://xxxxxxxxx.sharepoint.com/sites/URLSite/FormServerTemplates 
-Title:"XXXX XXXX XXXX"
-filetype:aspx

**Comment by KaveSys** [comment]
Hello @kasperbolarsen,
Let me explain better.

I have a link that have in the address the word "AV3BINSA".
If I try to search for example the "AV3BINSA" word in Search Box, I want that in Search Results the link with the word "AV3BINSA" will be visible in the results:

![image](https://github.com/user-attachments/assets/a2a4e739-f4de-4325-9bd2-b1555e790de3)


![image](https://github.com/user-attachments/assets/3a1eebd2-14ca-4ca0-818c-768f918ff286)

![image](https://github.com/user-attachments/assets/a8830310-cd0a-4a3d-9e25-6b8ed1590016)

In example, this is a link set in a Quick Link web part that go to a pdf file placed in another Sharepoint Online website.

I hope that now is more clear :)

**Comment by kasperbolarsen** [comment]
Nope, I still don't understand the issue, but it sounds like you are using an approach like this one https://microsoft-search.github.io/pnp-modern-search/scenarios/use-query-string-in-url/  when the query text is in the URL, and as per the screen shorts that part is working fine, right?

**Comment by wobba** [comment]
You should change `contentclass:STS_ListItem` to `contentclass:STS_ListItem_*` and similar for other contentclass filters where you are adding the prefix.

That said, the quick links webpart store links inside the web part properties afaik, which means you cannot easily search for them and have them displayed. The will be part of the content on the page of the web part, assuming the web part is set up to allow indexing of the links.
